### DISCUSSION TITLE
warning message/logged event of factor levels.

### DISCUSSION THREAD

**Discussion by Michael-Covell** [amices/mice#352] [Missing data methodology]
I'm getting a warning message/logged event that returns in the output field *some* of a factor variable's factor levels.  I get this type of logged event whenever I include factor variables.  What does this mean?

[xyz$loggedEvents3.txt](https://github.com/amices/mice/files/5126675/xyz.loggedEvents3.txt)

**Comment by stefvanbuuren** [comment]
```
it	im	dep	meth	out
1	1	broilers_boosted	norm	"menu_aggMeat, menu_aggNA, menu_aggPizza"
1	2	broilers_boosted	norm	"menu_aggMeat, menu_aggNA, menu_aggPizza"
2	1	broilers_boosted	norm	"menu_aggMeat, menu_aggNA, menu_aggPizza"
2	2	broilers_boosted	norm	"menu_aggMeat, menu_aggNA, menu_aggPizza"
3	1	broilers_boosted	norm	"menu_aggMeat, menu_aggNA, menu_aggPizza"
3	2	broilers_boosted	norm	"menu_aggMeat, menu_aggNA, menu_aggPizza"
4	1	broilers_boosted	norm	"menu_aggMeat, menu_aggNA, menu_aggPizza"
4	2	broilers_boosted	norm	"menu_aggMeat, menu_aggNA, menu_aggPizza"
5	1	broilers_boosted	norm	"menu_aggMeat, menu_aggNA, menu_aggPizza"
5	2	broilers_boosted	norm	"menu_aggMeat, menu_aggNA, menu_aggPizza"
```

Variables `menu_aggMeat`, `menu_aggNA` and `menu_aggPizza` are removed from the imputation model for `broilers_boosted` because of linear dependencies among the predictors. Remove these variables using the `predictorMatrix`. 

See https://stefvanbuuren.name/fimd/sec-toomany.html#finding-problems-loggedevents

**Comment by Michael-Covell** [comment]
Thanks, `menu_aggMeat`, `menu_aggNA` and `menu_aggPizza` are not variables though.  They are levels of a categorical variable.  Do you mean that these 3 levels are removed and the others (not listed) are used in imputations?  If so, then removing variable `menu_agg` from the `predictorMatrix` could remove information from the model.

**Comment by stefvanbuuren** [comment]
Yes, `mice` removes the dummies variables for these categories. 

There might be, for example, another categorical predictor with the same cases in its category. In that case, you may end up with two identical dummy variables. One needs to leave.
